moto/cognitoidp/utils.py

```diff
@@ -4,6 +4,12 @@
 import hashlib
 import hmac
 import base64
+import re
+
+FORMATS = {
```

Comment:
Are these regexes defined by AWS somewhere? I'm a little worried that AWS will have something slightly different, and we'll make it too restrictive

Reply:
I've only been able to find documentation regarding the phone number: 

> Phone numbers must follow these formatting rules: A phone number must start with a plus (+) sign, followed immediately by the country code. A phone number can only contain the + sign and digits. You must remove any other characters from a phone number, such as parentheses, spaces, or dashes (-) before submitting the value to the service. For example, a United States-based phone number must follow this format: +14325551212.

[Source - under `phone_number`](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-usernames)

Regarding the email address, I pulled the regex from [here](https://www.geeksforgeeks.org/check-if-email-address-valid-or-not-in-python/)

Reply:
Fair enough, that should be OK